internal/services/appservice/helpers/function_app_slot_schema.go

```diff
@@ -915,27 +888,41 @@ func FlattenSiteConfigWindowsFunctionAppSlot(functionAppSlotSiteConfig *web.Site
 	}
 
 	if functionAppSlotSiteConfig.Cors != nil {
+		corsEmpty := false
 		corsSettings := functionAppSlotSiteConfig.Cors
 		cors := CorsSetting{}
 		if corsSettings.SupportCredentials != nil {
 			cors.SupportCredentials = *corsSettings.SupportCredentials
 		}
 
-		if corsSettings.AllowedOrigins != nil && len(*corsSettings.AllowedOrigins) != 0 {
-			cors.AllowedOrigins = *corsSettings.AllowedOrigins
+		if corsSettings.AllowedOrigins != nil {
+			if len(*corsSettings.AllowedOrigins) > 0 {
+				cors.AllowedOrigins = *corsSettings.AllowedOrigins
+			} else if !cors.SupportCredentials {
+				corsEmpty = true
+			}
+		}
+		if !corsEmpty {
+			result.Cors = []CorsSetting{cors}
 		}
-		result.Cors = []CorsSetting{cors}
 	}
 
-	var appStack []ApplicationStackWindowsFunctionApp
-	if functionAppSlotSiteConfig.WindowsFxVersion != nil {
-		decoded, err := DecodeFunctionAppWindowsFxVersion(*functionAppSlotSiteConfig.WindowsFxVersion)
-		if err != nil {
-			return nil, fmt.Errorf("flattening site config: %s", err)
+	powershellVersion := ""
+	if p := functionAppSlotSiteConfig.PowerShellVersion; p != nil {
+		powershellVersion = *p
+		if powershellVersion == "~7" {
+			powershellVersion = "7"
 		}
-		appStack = decoded
 	}
-	result.ApplicationStack = appStack
+
+	result.ApplicationStack = []ApplicationStackWindowsFunctionApp{{
+		DotNetVersion:         pointer.From(functionAppSlotSiteConfig.NetFrameworkVersion),
+		DotNetIsolated:        false, // set this later from app_settings
+		NodeVersion:           "",    // Need to get this from app_settings later
```

Comment:
Is this a todo?

Reply:
No, it's a hint that we're setting an empty value here and picking up the actual value in a later step. I'll update the comment to make this more clear.